vnext/Directory.Build.props

```diff
@@ -24,6 +21,22 @@
 
   <PropertyGroup Label="Configuration">
     <ProjectName Condition="'$(ProjectName)'==''">$(MSBuildProjectName)</ProjectName>
+
+    <RootIntDir Condition="'$(RootIntDir)' == ''">$(ReactNativeWindowsDir)build</RootIntDir>
+    <RootOutDir Condition="'$(RootOutDir)' == ''">$(ReactNativeWindowsDir)target</RootOutDir>
+    <!-- Visual Studio forces using 'Win32' for the 'x86' platform. -->
+    <BaseIntDir Condition="'$(BaseIntDir)' == '' AND '$(Platform)' == 'Win32'">$(RootIntDir)\x86\$(Configuration)</BaseIntDir>
+    <BaseIntDir Condition="'$(BaseIntDir)' == '' AND '$(Platform)' != 'Win32'">$(RootIntDir)\$(Platform)\$(Configuration)</BaseIntDir>
+    <BaseOutDir Condition="'$(BaseOutDir)' == '' AND '$(Platform)' == 'Win32'">$(RootOutDir)\x86\$(Configuration)</BaseOutDir>
+    <BaseOutDir Condition="'$(BaseOutDir)' == '' AND '$(Platform)' != 'Win32'">$(RootOutDir)\$(Platform)\$(Configuration)</BaseOutDir>
+
+    <IntDir Condition="'$(ProjectName)' != ''">$(BaseIntDir)\$(ProjectName)\</IntDir>
+    <IntDir Condition="'$(ProjectName)' == ''">$(BaseIntDir)\$(MSBuildProjectName)\</IntDir>
+    <OutDir Condition="'$(ProjectName)' != ''">$(BaseOutDir)\$(ProjectName)\</OutDir>
+    <OutDir Condition="'$(ProjectName)' == ''">$(BaseOutDir)\$(MSBuildProjectName)\</OutDir>
+
+    <PublishDir>$(OutDir)\Publish\</PublishDir>
+    <GeneratedFilesDir>$(IntDir)Generated Files\</GeneratedFilesDir>
```

Comment:
Why do we need to add these back?

Reply:
One of the new properties in this PR used `RootOutDir`.
I couldn't track how exactly these were replaced, so just restored them.

They were originally meant to provide build flexibility. Do they overlap with another relevant property?

Reply:
Direct usage of `$(RootIntDir)` is not friendly to parallel project evaluation. The last build target using it was seeing contention issues, where the same target in multiple projects tried to write to the same file. The built-in `$(IntDir)` is per-project, to partition their outputs from each other. If we publicly expose an ability to write above that, we break that isolation.